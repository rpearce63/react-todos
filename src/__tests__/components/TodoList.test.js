import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";

import ConnectedTodoList, { TodoList } from "../../components/TodoList";

describe("TodoList component", () => {
  const props = {
    todos: [
      {
        id: 1,
        text: "my todo",
        complete: false
      },
      {
        id: 2,
        text: "my other todo",
        complete: true
      }
    ]
  };
  const wrapper = shallow(<TodoList {...props} />);

  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should contain a list of Todos", () => {
    expect(wrapper.find("Connect(Todo)")).toHaveLength(2);
  });

  it("should render the connected coomponent properly", () => {
    const mockStore = configureStore();
    const store = mockStore({ ...props });
    const connected = shallow(<ConnectedTodoList store={store} />);
    expect(connected.exists()).toBe(true);
    expect(connected.prop("todos")).toEqual(props.todos);
  });
});
