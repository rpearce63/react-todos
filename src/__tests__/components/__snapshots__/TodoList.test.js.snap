// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoList
    todos={
      Array [
        Object {
          "complete": false,
          "id": 1,
          "text": "my todo",
        },
        Object {
          "complete": true,
          "id": 2,
          "text": "my other todo",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Todos
        </h3>,
        Array [
          <Connect(Todo)
            todo={
              Object {
                "complete": false,
                "id": 1,
                "text": "my todo",
              }
            }
          />,
          <Connect(Todo)
            todo={
              Object {
                "complete": true,
                "id": 2,
                "text": "my other todo",
              }
            }
          />,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Todos",
        },
        "ref": null,
        "rendered": "Todos",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "todo": Object {
            "complete": false,
            "id": 1,
            "text": "my todo",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "todo": Object {
            "complete": true,
            "id": 2,
            "text": "my other todo",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Todos
          </h3>,
          Array [
            <Connect(Todo)
              todo={
                Object {
                  "complete": false,
                  "id": 1,
                  "text": "my todo",
                }
              }
            />,
            <Connect(Todo)
              todo={
                Object {
                  "complete": true,
                  "id": 2,
                  "text": "my other todo",
                }
              }
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Todos",
          },
          "ref": null,
          "rendered": "Todos",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "todo": Object {
              "complete": false,
              "id": 1,
              "text": "my todo",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "todo": Object {
              "complete": true,
              "id": 2,
              "text": "my other todo",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
